### ISSUE TITLE
Testnet becomes unusable after large hashrate drop

### ISSUE THREAD

**Issue by garethtdavies** [zcash/zcash#3552]
Since the removal of the testnet-only difficulty rules, which as per Bitcoin would allow mining of a min-difficulty block after 2 * 2.5 mins (https://github.com/zcash/zcash/commit/333ea3c4266179da5d40e496ae60bcce0259c790) the testnet doesn't have an easy way to recover from a dramatic drop in hashrate and can lead to extended periods of the testnet not finding blocks.

Currently, the testnet is stuck in such a state with the network hashrate at over 60k Sol/s (which has obviously long-since departed) where it previously hovered around ~20 Sol/s.

```
{
"blocks": 299179,
"currentblocksize": 0,
"currentblocktx": 0,
"difficulty": 262594.272944737,
"errors": "WARNING: check your network connection, 0 blocks received in the last 4 hours (96 expected)",
"genproclimit": 1,
"localsolps": 0,
"networksolps": 60783,
"networkhashps": 60783,
"pooledtx": 0,
"testnet": true,
"chain": "test",
"generate": false
}
```

**Comment by daira** [comment]
I agree something must be done promptly to fix the problem on testnet, in order for it to be usable for Sapling testing.

As noted by @str4d in the commit comment for 333ea3c4266179da5d40e496ae60bcce0259c790, it's actually tricky to apply the Bitcoin testnet's min-difficulty rule to Zcash due to the change in difficulty adjustment algorithm.

We also need to consider compatibility issues; it's acceptable to cause a testnet fork (since it's currently unusable), but we need to be sure that nodes that upgrade after the fork will correctly reorg to the new chain, which is not obvious when the min-difficulty rule is involved if the node has seen blocks after the fork on the old chain.

**Comment by daira** [comment]
Arguably, the min-difficulty time threshold should be set so that min-difficulty blocks occur rarely. For twice the block time they occur frequently due to the exponential distribution of interarrival times. I suggest 6 times the block time, or 15 minutes. (This is independent of the other issues in my previous comment.)

**Comment by daira** [comment]
@str4d and I simulated the effect of large (99%) drops in hash power on a Zcash-like testnet, with no min-difficulty rule and with a min-difficulty threshold time set to 2, 4, 6, or 20 times the target block time. My intuition in the previous comment was correct: 6 times the target block time (i.e. 15 minutes) is long enough to make min-difficulty blocks infrequent in normal operation, but allow recovery from a large hash power drop within a few hours.

This did not seem to cause any significant problem with chain splits or orphans (the simulation is of a network of 100 nodes; and takes into account block propagation and reorgs; we had to modify the calculation of chain work up make it more realistic). I'll add graphs of the results later.
